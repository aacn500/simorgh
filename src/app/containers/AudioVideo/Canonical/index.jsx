import React from 'react';
import styled from 'styled-components';
import { string, bool } from 'prop-types';

const Canonical = ({ id, portrait }) => {
  const position = portrait ? 'position: absolute;' : '';

  const PortraitParent = styled.div`
    padding-bottom: 178.78%;
    position: relative;
  `;

  const MediaPlayerContainer = styled.div`
    height: 100%;
    width: 100%;
    ${position}
  `;

  if (portrait) {
    return (
      <PortraitParent>
        <MediaPlayerContainer id={id} key={id} />
      </PortraitParent>
    );
  }

  return <MediaPlayerContainer id={id} key={id} />;
};

Canonical.propTypes = {
  id: string.isRequired,
  portrait: bool.isRequired,
};

export default Canonical;
