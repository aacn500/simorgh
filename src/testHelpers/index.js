import renderer from 'react-test-renderer';
import 'jest-styled-components';
import ShallowRenderer from 'react-test-renderer/shallow';
import deepClone from 'ramda/src/clone';
import nodeLogger from '../app/lib/logger.node';

export const shouldMatchSnapshot = (title, component) => {
  it(title, () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export const shallowRender = component => {
  const shallowRenderer = new ShallowRenderer();
  shallowRenderer.render(component);
  const result = shallowRenderer.getRenderOutput();

  return result;
};

export const shouldShallowMatchSnapshot = (title, component) => {
  it(title, () => {
    const tree = shallowRender(component);
    expect(tree).toMatchSnapshot();
  });
};

export const isNull = (title, component) => {
  it(title, () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toBeNull();
  });
};

export const setWindowValue = (key, value) => {
  const windowValue = window[key];
  delete window[key];

  let newValue = value;

  if (value && typeof value === 'object') {
    newValue = {
      ...deepClone(windowValue),
      ...value,
    };
  }

  Object.defineProperty(window, key, {
    value: newValue,
    writable: true,
  });
};

export const resetWindowValue = (key, value) => {
  Object.defineProperty(window, key, {
    value,
    writable: true,
  });
};

const mocks = {
  error: jest.fn(),
  warn: jest.fn(),
  info: jest.fn(),
  verbose: jest.fn(),
  debug: jest.fn(),
  silly: jest.fn(),
};

jest.mock('../app/lib/logger.node', () => jest.fn());
nodeLogger.mockImplementation(() => mocks);

export const loggerMock = mocks;
