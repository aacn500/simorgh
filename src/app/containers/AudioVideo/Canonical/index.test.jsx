import React from 'react';
import { shouldMatchSnapshot } from '../../../../testHelpers';
import Canonical from '.';

describe('Canonical', () => {
  const examplePID = 'p01k6msm';

  shouldMatchSnapshot(
    'should place Canonical container (landscape video) on page',
    <Canonical id={examplePID} portrait={false} />,
  );

  shouldMatchSnapshot(
    'should place Canonical container (portrait video) on page',
    <Canonical id={examplePID} portrait />,
  );
});
