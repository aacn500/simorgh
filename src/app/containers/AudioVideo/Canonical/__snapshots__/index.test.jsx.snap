// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canonical should place Canonical container (landscape video) on page 1`] = `
.c0 {
  height: 100%;
  width: 100%;
}

<div
  className="c0"
  id="p01k6msm"
/>
`;

exports[`Canonical should place Canonical container (portrait video) on page 1`] = `
.c0 {
  padding-bottom: 178.78%;
  position: relative;
}

.c1 {
  height: 100%;
  width: 100%;
  position: absolute;
}

<div
  className="c0"
>
  <div
    className="c1"
    id="p01k6msm"
  />
</div>
`;
